packages/contracts/src/dollar/core/BancorFormula.sol

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.3;
+
+// import "./interfaces/IBancorFormula.sol";
+import "./BancorPower.sol";
+import "abdk/ABDKMathQuad.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+
+/**
+ * @title Bancor formula
+ * @dev Modified from the original
+ * @notice forked from Bancor:
+ * https://github.com/bancorprotocol/contracts
+ *
+ */
+contract BancorFormula is BancorPower {
+    using ABDKMathQuad for uint256;
+    using ABDKMathQuad for bytes16;
+
+    bytes16 private immutable ONE = (uint256(1)).fromUInt();
+    uint32 private constant MAX_WEIGHT = 1e6;
+
+    /**
+     * @dev Given a token supply, reserve balance, weight and a deposit amount (in the reserve token),
+     * calculates the target amount for a given conversion (in the main token)
+     *
+     * @dev _supply * ((1 + _tokensDeposited / _connectorBalance) ^ (_connectorWeight / 1000000) - 1)
+     *
+     * @param _tokensDeposited   amount of collateral tokens to deposit
+     * @param _connectorWeight   connector weight, represented in ppm, 1 - 1,000,000
+     * @param _supply          current Token supply
+     * @param _connectorBalance   total connector balance
+     * 
+     * @return amount of Tokens minted
+     */
+    function _purchaseTargetAmount(
```

Comment:
I think we can remove this contract as it is in the facte now

Reply:
alright